pkg/tsdb/postgres/macros.go

```diff
@@ -15,12 +16,15 @@ const rsIdentifier = `([_a-zA-Z0-9]+)`
 const sExpr = `\$` + rsIdentifier + `\(([^\)]*)\)`
 
 type postgresMacroEngine struct {
-	timeRange *tsdb.TimeRange
-	query     *tsdb.Query
+	timeRange   *tsdb.TimeRange
+	query       *tsdb.Query
+	timescaledb bool
 }
 
-func newPostgresMacroEngine() tsdb.SqlMacroEngine {
-	return &postgresMacroEngine{}
+func newPostgresMacroEngine(datasource *models.DataSource) tsdb.SqlMacroEngine {
```

Comment:
Would make it more explicit and easier to test if you accept the boolean as argument. In addition, the postgresMacroEngine wont have any dependencies to models.DataSource and don't need to understand the models.DataSource model - that dependency can live in postgres.go. By this change you don't need to create the models.DataSource over and over again in both macros_test and postgres_test.

Make sense?

Reply:
newPostgresMacroEngine(false) looks a bit weird but I agree

Reply:
Yes agree. But `newPostgresMacroEngine(timescaledbEnabled)` looks better.